### ISSUE TITLE
Simulated reads error rate

### ISSUE THREAD

**Issue by AntonS-bio** [ZKI-PH-ImageAnalysis/seq2squiggle#17]
I'm trying to finetune the simulated reads to reflect the error rate I get in the real data (~1.5% per base), but after testing multiple parameters the best I got was 13% error rate using dorado dna_r10.4.1_e8.2_400bps_sup@v5.0.0 model . As I can't train a new model to improve accuracy, can you please let me know if it's possible to achieve the 1.5% error rate by tweaking some parameters of the existing models? Thank you.

**Comment by denisbeslic** [comment]
Could you please share the full command and parameters you use to generate reads with seq2squiggle?
For our evaluation, we basecalled the R10-4kHz reads using dna_r10.4.1_e8.2_400bps_sup@v4.1.0 and the R10-5kHz reads using dna_r10.4.1_e8.2_400bps_sup@v5.0.0.
For simulation, we used the following parameters:
`-n [number_of_reads] --profile dna_r10_min --duration-sampler True --noise-sampler True -r 3000 --sample-rate 4000`

Please note that for simulating R10-5kHz data, you need to adjust the --sample-rate accordingly (change it to 5000).

**Comment by denisbeslic** [comment]
Thanks for the files! I'm currently out of office but will take a look next week.

Regarding the read length inconsistency, just to clarify: by default, `seq2squiggle` expects a genome as input and first generates reads by sampling from it. The `--read-length` parameter specifies an average read length, but the actual lengths can vary (i.e., some reads may be shorter or longer). If your reference sequence is only 820 bases long, for example, the reads will naturally be shorter than the specified average. Based on those samples reads, it will generate the signals you find in the pod5 file.

Alternatively, you can use the `--read-input` parameter. In this mode, `seq2squiggle` treats the input as reads and simulates the entire read as-is, without further sampling. Currently, this mode generates only 1 signal per sequence, but I can adjust this.

Would you like the reference amplicon sequence to be simulated in full, or should it be split further?

**Comment by denisbeslic** [comment]
@AntonS-bio 
I took a closer look at the data, and one likely source of error appears to be the genome mode vs. read mode confusion I mentioned in my previous comment. I’ve now added support for generating multiple signals from the input data to the dev branch and will update the README shortly to clarify this behavior.

We’re currently working on refining the noise profile to better match R10 data. As noted in our original paper, there’s still a discrepancy between experimental data (error rate <1%) and simulated data (2.5–4.5% in our evaluated datasets) when using R10 chemistry.

To improve the mismatch rate for now, you can disable the duration sampler (which makes all events the same length) and slightly increase the noise level using --noise-std 2.0. See also [Supplementary Figure 12 ](https://oup.silverchair-cdn.com/oup/backfile/Content_public/Journal/bioinformatics/41/1/10.1093_bioinformatics_btae744/1/btae744_supplementary_data.docx?Expires=1751310450&Signature=yqUt6fgtQAWCXPABtRRB-7LC9qrU-IB0lcy56DFtX8SHuOLnLhLtfBzEGNloYxBAuxrW17O7J~yqdYATBcKNH6FKGHaqPAx9DQLzrch-h0nl4LKLHOoqJfInLcRDlsRQNmw6JmeTPmlZzicyb9~ypmcHffmUw7Ja5tzikd1GcYeOAWp0Qex0--Ju2SlZTeLE4lSMAR3K~kM7s4E6Spi3sv8chp4YqtDlUSpBqbykTmcjlyDsLxbZctAYXAU5muyfwUTtxpXLK7xcBYZeyuSaIhBYkbUx7kizKTRlAZbHgVAkicIdIiDfkDCV1T9PauHeXnzfuXgqxgxWl0XzMrcw8g__&Key-Pair-Id=APKAIE5G5CRDK6RD3PGA )in the paper for reference.

Using the latest commit from the dev branch, I was able to achieve an average error rate of 1.7% with the following commands:
```bash
seq2squiggle predict amplicon-data/amplicon.fna -o amplicon-data/out-fullreads50.pod5 -n 50 -r -1 --duration-sampler False --noise-std 2.0
dorado basecaller dna_r10.4.1_e8.2_400bps_sup@v5.0.0 amplicon-data/out-fullreads50.pod5 --emit-fastq > amplicon-data/out-fullreads.fastq
minimap2 -ax map-ont amplicon-data/amplicon.fna amplicon-data/out-fullreads.fastq | samtools stats - | grep ^SN | grep 'error rate'
```

I’ve attached the resulting data for your reference. 
[out-fullreads.fastq.txt](https://github.com/user-attachments/files/20445474/out-fullreads.fastq.txt)
[out-fullreads50.pod5.txt](https://github.com/user-attachments/files/20445475/out-fullreads50.pod5.txt)

The changes to seq2squiggle, including support for simulating multiple signals for a full reference read, will be merged into the main branch and released soon.

Let me know if this helps resolve the issue!
